Dependency mismatch in `v9.0.0-rc.1`

### Environment

- Operating System: `Darwin`
- Node Version:     `v20.17.0`
- Nuxt Version:     `3.13.2`
- CLI Version:      `3.14.0`
- Nitro Version:    `2.9.7`
- Package Manager:  `npm@10.8.2`
- Builder:          `-`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`


### Reproduction

On the local environment and on the stackblitz I have different errors, but both envs not working for me

On stackblitz I have `Pre-transform error: compiler is null (x2)`
https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-rt9kda?file=i18n%2Fi18n.config.ts

On the local I have
```
 [ERROR] No matching export in "node_modules/@intlify/shared/dist/shared.mjs" for import "incrementer"

    node_modules/@intlify/core-base/dist/core-base.mjs:6:121:
      6 │ ...mber, isPlainObject, assign, join, toDisplayString, isArray, incrementer, format as format$1, isPromise, isBoolean, warn, isRegExp, ...
        ╵                                                                 ~~~~~~~~~~~


 ERROR  error while updating dependencies:                                                                                             11:05:53 AM
Error: Build failed with 1 error:
node_modules/@intlify/core-base/dist/core-base.mjs:6:121: ERROR: No matching export in "node_modules/@intlify/shared/dist/shared.mjs" for import "incremented"

```

### Describe the bug

v9 rc1 not run

### Additional context

`npm list @intlify/shared` gives

```
└─┬ @nuxtjs/i18n@9.0.0-rc.1
  ├─┬ @intlify/h3@0.5.0
  │ └─┬ @intlify/core@9.14.1
  │   ├─┬ @intlify/core-base@9.14.1
  │   │ └── @intlify/shared@9.14.1
  │   └── @intlify/shared@9.14.1
  ├── @intlify/shared@10.0.3
  ├─┬ @intlify/unplugin-vue-i18n@5.2.0
  │ ├─┬ @intlify/bundle-utils@9.0.0-beta.0
  │ │ ├─┬ @intlify/message-compiler@9.14.1
  │ │ │ └── @intlify/shared@9.14.1
  │ │ └── @intlify/shared@10.0.3 deduped
  │ ├── @intlify/shared@10.0.3 deduped
  │ └─┬ @intlify/vue-i18n-extensions@7.0.0
  │   └── @intlify/shared@10.0.3 deduped
  └─┬ vue-i18n@10.0.3
    ├─┬ @intlify/core-base@10.0.3
    │ ├─┬ @intlify/message-compiler@10.0.3
    │ │ └── @intlify/shared@10.0.3 deduped
    │ └── @intlify/shared@10.0.3 deduped
    └── @intlify/shared@10.0.3 deduped
```

probably the issue with intlify/h3 which requires intlify/shared@9

### Logs

_No response_

--- a/package.json
+++ b/package.json
@@ -84,7 +84,7 @@
     }
   },
   "dependencies": {
-    "@intlify/h3": "^0.5.0",
+    "@intlify/h3": "^0.6.0",
     "@intlify/shared": "^10.0.3",
     "@intlify/unplugin-vue-i18n": "^5.2.0",
     "@intlify/utils": "^0.13.0",

